File: ApplicationProjectRW1/subgraphs/sg1/code/src/main/scala/simpledatalabs_27/pm3rework/subgraph/sg1/pipelines/pipeline1/graph/Subgraph_1/Reformat_2.scala
package simpledatalabs_27.pm3rework.subgraph.sg1.pipelines.pipeline1.graph.Subgraph_1

import io.prophecy.libs._
import io.prophecy.pipelines.pipeline1.config.ConfigStore._
import io.prophecy.pipelines.pipeline1.config.Context
import io.prophecy.pipelines.pipeline1.udfs.UDFs._
import io.prophecy.pipelines.pipeline1.udfs._
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object Reformat_2 {

  def apply(context: Context, in: DataFrame): DataFrame =
    in.select(trimUDF(col("customer_id")).as("customer_id"))

}

File: baseProjectPY/subgraphs/sg1/code/sg1/Subgraph_1/Reformat_2.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.libs import typed_lit
from .config import *
from basepipeline1.udfs.UDFs import *

def Reformat_2(spark: SparkSession, in0: DataFrame) -> DataFrame:
    return in0.select(
        udfConcat(col("first_name"), col("last_name")).alias("name"), 
        udfConcatold(col("first_name"), col("last_name")).alias("nameold")
    )
